nox/manifest.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2017 Jon Wayne Parrott
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+import copy
+import itertools
+
+from nox._parametrize import generate_calls
+from nox.logger import logger
+from nox.sessions import Session
+
+
+class Manifest(object):
+    """Session manifest.
+
+    The session manifest provides the final list of sessions that should be
+    run by nox.
+
+    It is possible for this to be mutated during execution. This allows for
+    useful use cases, such as for one session to "notify" another.
```

Comment:
`notify` it will remain, unless we think of something better. :-) And yes, I agree with mostly pulling terminology from py.test, but too saw no candidate.